### PR TITLE
Add support for Django 2.0, 2.1 & 2.2

### PR DESCRIPTION
**Pull request by joehybird**

### CODE REVIEW

**Comment by Natim** [comment]
I guess we should drop support for the red ones here:

image

File: .travis.yml [hunk 1]
```diff
@@ -1,36 +1,22 @@
...
-        - python: 3.4
-          env: TOX_ENV=py34-django111
+dist: xenial
+sudo: true
+services:
+  - postgresql
+python:
+  - 2.7
+  - 3.5
+  - 3.6
```
Comment (Natim) [inline .travis.yml]: Would you consider adding support for Python 3.7 and Python 3.8?

File: .travis.yml [hunk 2]
```diff
@@ -1,36 +1,22 @@
...
 
-        - python: 2.7
-          env: TOX_ENV=flake8
-script:
-    - tox -e $TOX_ENV
-install:
-    pip install tox
+jobs:
+  include:
+    - { stage: lint, env: TOXENV=flake8, python: 3.6 }
```
Comment (Natim) [inline .travis.yml]: :+1:

File: setup.py [hunk 3]
```diff
@@ -15,7 +15,7 @@ def read_relative_file(filename):
 VERSION = read_relative_file('VERSION')
 PACKAGES = ['django_genericfilters']
 REQUIRES = [
-    'Django>=1.8,<1.12',
+    'Django>=1.8,<3.0',
```
Comment (Natim) [inline setup.py]: Why not support Django 3.0?

**Comment by Natim** [comment]
I would target python3.8 rather than python3.6 for the stage and lint or basepython config.

File: setup.py (hunk 3 above)
Comment (joehybird) [reply setup.py]: 2.2 is the LTS so I only needed this version. If 3.0 does not cause compatibility issues, why not :)

Comment (joehybird) [reply setup.py]: Done :)

File: .travis.yml (hunk 1 above)
Comment (joehybird) [reply .travis.yml]: Done :)

File: demo/demoproject/settings.py [hunk 4]
```diff
@@ -75,13 +79,22 @@
...
+        'django.contrib.messages.middleware.MessageMiddleware',
+    ]
+else:
+    MIDDLEWARE_CLASSES = [
+        'django.middleware.common.CommonMiddleware',
+        'django.contrib.sessions.middleware.SessionMiddleware',
+        'django.middleware.csrf.CsrfViewMiddleware',
+        'django.contrib.auth.middleware.AuthenticationMiddleware',
+        'django.contrib.messages.middleware.MessageMiddleware',
+    ]
```
Comment (k4nar) [inline demo/demoproject/settings.py]: I think this can be removed now :) .

Comment (joehybird) [reply demo/demoproject/settings.py]: Arf, you have a point :)

**Comment by joehybird** [comment]
Anyone can merge this ? :smile:

**Comment by joehybird** [comment]
Anyone can RELEASE it ? :sweat_smile:

**Pull request merged**